prow/plank/controller.go

```diff
@@ -180,6 +181,12 @@ func (c *Controller) syncJenkinsJob(pj kube.ProwJob) error {
 	if pj.Complete() {
 		return nil
 	} else if pj.Status.State == kube.TriggeredState {
+		if pj.Spec.MaxConcurrency > 0 && len(pjm[pj.Spec.Job]) >= pj.Spec.MaxConcurrency {
+			// do not start more jobs than allowed
+			// TODO(nclandolfi): log?
+			return nil
+		}
+
 		// Start the Jenkins job.
 		pj.Status.State = kube.PendingState
```

Comment:
Shouldn't we allow the status to be updated to pending? Also in your filter, you should probably include TriggeredState PJs.